### ISSUE TITLE
Remove flakiness from scan tests

### ISSUE THREAD

**Issue by mijicd** [zio/zio-redis#212]
All scan commands accepts count of the elements to retrieve. However, according to the official docs that count might be ignored:

```
In the COUNT option documentation, we state that sometimes this family of commands may return all the elements of a Set, Hash or Sorted Set at once in a single call, regardless of the COUNT option value.
```

To avoid flaky tests [like these](https://app.circleci.com/pipelines/github/zio/zio-redis/447/workflows/2c18f1a2-08da-4a54-8f73-c93a468d15cf/jobs/2653), I propose to rewrite them in a way that would repeat the scan until the returned cursor is zero, and assert the result set.

**Comment by justinhj** [comment]
yeah i was worried about this. the count is a hint 🤪

**Comment by justinhj** [comment]
I'll make a start on this

**Comment by mijicd** [comment]
All yours, thanks!

**Comment by mijicd** [comment]
Closed in #215
